### PR TITLE
PDO bind parameters support named parameters

### LINKED ISSUE #880: Forcing PDO bind parameter to be numeric array breaks application
# Problems summary

Pull Request #706 forces PDO bind parameter to be numeric array, but this makes named parameters unusable and some applications may not work.

Issue #705 says that the PDO driver requires numeric array, but in fact it seems that it is only required when the SQL contains a question mark. [The documentation](https://www.php.net/manual/en/pdo.prepare.php#example-1040) also says that named parameters are available.

## Expected

ADOdb with PDO driver allows named parameters.
(It means that I can execute SQL like in [this comment](https://github.com/ADOdb/ADOdb/issues/705#issuecomment-1144553805))

## Environment Information 

(This is the environment I tested)

- PHP 7.4 / 8.0 (with PDO MySQL driver)
- MySQL 5.7 / 8.0
- ADOdb 5.22.3
  - probably occurs in 5.21.1 and later

# How to reproduce the problem

- Here is a sample code showing that PHP's PDO Driver can use named parameters.
  - https://github.com/castor4bit/pdo-driver-allows-named-parameters
  - Tests using PDO directly pass, fail with ADOdb

### PR DESCRIPTION
**Pull request by castor4bit**
Fix #880

### CODE REVIEW

**Comment by castor4bit** [comment]
```
$ ./vendor/bin/phpunit tests/PdoDriverTest.php
PHPUnit 8.5.30 #StandWithUkraine

..........................                                        26 / 26 (100%)

Time: 558 ms, Memory: 4.00 MB

OK (26 tests, 26 assertions)
```

**Review by dregad — APPROVED** [review]
Looks good to me, except for the typo

**Comment by dregad** [comment]
Having read the discussion in #880, it seems you guys have reached consensus, and this PR seems ready to be merged.

@mnewnham let me know if there are any objections.

File: drivers/adodb-pdo.inc.php [hunk 1]
```diff
@@ -92,6 +96,15 @@ class ADODB_pdo extends ADOConnection {
 	*/
 	public $pdoParameters = array();
 
+	/*
+	* Set witch style is used to bind parameters
```
[the remark this answers is not in the document]
Comment (castor4bit) [reply drivers/adodb-pdo.inc.php]: Thanks for pointing out. I fixed it.

**Comment by dregad** [comment]
@mnewnham OK if I merge this ? Target 5.23.0 ?

**Comment by mnewnham** [comment]
> @mnewnham OK if I merge this ? Target 5.23.0 ?

Yes that's fine thanks

**Comment by dregad** [comment]
@castor4bit many thanks for your contribution, and apologies for taking so long to merge it.

**Pull request merged**